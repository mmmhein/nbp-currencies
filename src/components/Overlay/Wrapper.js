import styled from 'styled-components';

import { spacingLarge } from '../../styles/designTokens';

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  margin: ${spacingLarge};
`;

export default Wrapper;
