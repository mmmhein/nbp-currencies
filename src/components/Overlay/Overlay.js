import React from 'react';
import PropTypes from 'prop-types';

import Text from '../Text';
import Wrapper from './Wrapper';

function Overlay({ children }) {
  return (
    <Wrapper data-testid="overlay">
      <Text bold>{children}</Text>
    </Wrapper>
  );
}

Overlay.propTypes = {
  children: PropTypes.node
};

export default Overlay;
