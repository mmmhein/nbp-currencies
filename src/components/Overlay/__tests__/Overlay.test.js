import React from 'react';
import { create } from 'react-test-renderer';

import Overlay from '../Overlay';

jest.mock('../../Text', () => 'Text');

describe('COMPONENT - Overlay', () => {
  it('renders correctly', () => {
    const component = create(<Overlay />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
