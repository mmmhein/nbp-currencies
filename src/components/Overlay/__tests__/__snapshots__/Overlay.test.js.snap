// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - Overlay renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 3rem;
}

<div
  className="c0"
  data-testid="overlay"
>
  <Text
    bold={true}
  />
</div>
`;
